### ISSUE TITLE
TestBPF/tc_nodeport_lb4_nat_lb.o: bpf_test.go:174: verifier error: load program: invalid argument: Validating mock_fib_lookup() func#1 arg#0 reference type('UNKNOWN ') size cannot be determined: -22

### ISSUE THREAD

**Issue by jingyuanliang** [cilium/cilium#35182]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.16.0 and lower than v1.17.0

### What happened?

bpf test failure with Clang 18

```
=== RUN   TestBPF/tc_nodeport_lb4_nat_lb.o
    bpf_test.go:174: verifier error: load program: invalid argument:
        	Validating mock_fib_lookup() func#1...
        	arg#0 reference type('UNKNOWN ') size cannot be determined: -22
        	processed 0 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0
```

### How can we reproduce the issue?

Run bpf tests with Clang 18 installed on the system using `make -C bpf/tests all`.

### Cilium Version

1.16.2

### Kernel Version

6.1.100+

### Kubernetes Version

N/A

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
Hi! The expectation when running the tests is that you use the same compiler as Cilium, with `make run_bpf_tests`. If you try that, do you still see the same problem?

**Comment by sayboras** [comment]
can you try again from main? This might be fixed recently with llvm 18 upgrade https://github.com/cilium/cilium/pull/36197

**Comment by sayboras** [comment]
This should be fixed as part of #36197.
